### PR TITLE
Arc admin attribute form

### PR DESCRIPTION
**Pull request by kbrownomni**
pulled arc-admin attributes form out of pop-up and into page. Updated html. Fixed tests.

### CODE REVIEW

File: plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts [hunk 1]
```diff
@@ -69,14 +72,79 @@ export class ArcAdminComponent implements OnInit {
     this.config = config;
     this.configChangedNotifier.next();
   }
-
+// Visually blank but has underlying value in editConfig
   ngOnInit(): void {
-  }
+    this.attributesForm = this.fb.group({
```
Comment (newmanw) [inline plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts:77]: This should be populated with the config returned from the server.  You can move this definition to the constructor and then populate the config with values returned from `fetchArcConfig(...)`

**Review by newmanw — CHANGES_REQUESTED** [review]
Update the UI to reflect changes pulled from server config.  All else looks good.

File: plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts [hunk 2]
```diff
@@ -69,14 +72,79 @@ export class ArcAdminComponent implements OnInit {
...
+      this.editConfig = this.copyConfig();
+      }
+      else{
+        console.log('Form is invalid, please correct the errors.')
+      }
+    }
+  
+  //when edit attributes is canceled, display and empty form but preserved data in editConfig
+  onCancel(): void {
+    this.attributesForm.reset({
```
Comment (Skosche3) [inline plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts:127]: Should reset empty these fields or reload from server side config?

File: plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts [hunk 3]
```diff
@@ -69,14 +72,79 @@ export class ArcAdminComponent implements OnInit {
...
+      usernameField: [''],
+      userDisplayNameField: [''],
+      deviceIdField: [''],
+      createdAtField: [''],
+      lastModifiedField: [''],
+      geometryType: ['']
+    });
+  }
+  
+  //Save attributes form. If left blank, assign default value
```
Comment (Skosche3) [inline plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts:93]: What will the default values be? From server config?

**Review by Skosche3 — COMMENTED** [review]
Looks good other than various places that should get values from server side config.

File: plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts (hunk 1 above)
Comment (kbrownomni) [reply plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts]: Okay, I think I did this correctly.

File: plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts (hunk 3 above)
Comment (kbrownomni) [reply plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts:119]: Yep! From the ArcGISPluginConfig file

File: plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts [hunk 4]
```diff
@@ -71,12 +89,95 @@ export class ArcAdminComponent implements OnInit {
   }
 
   ngOnInit(): void {
+    this.arcService.fetchArcConfig().pipe(first()).subscribe({
```
Comment (newmanw) [inline plugins/arcgis/web-app/projects/main/src/lib/arc-admin/arc-admin.component.ts:92]: When using angular httpClient you should not need to use `first() `

**Review by newmanw — APPROVED** [review]

**Pull request merged**